packages/eas-cli/src/api.ts

```diff
@@ -1,42 +1,96 @@
-        if (error instanceof HTTPError) {
-          let result: { [key: string]: any };
-          try {
-            result = JSON.parse(error.response.body as string);
-          } catch (e2) {
-            return error;
-          }
-          if (result.errors?.length) {
-            return new ApiV2Error(error, result.errors[0]);
-          }
-        }
-        return error;
-      },
-    ],
-  },
-});
+export class ApiV2Error extends Error {
+  readonly name = 'ApiV2Error';
+  readonly expoApiV2ErrorCode: string;
+  readonly expoApiV2ErrorDetails?: JSONValue;
+  readonly expoApiV2ErrorServerStack?: string;
+  readonly expoApiV2ErrorMetadata?: object;
+
+  constructor(response: {
+    message: string;
+    code: string;
+    stack?: string;
+    details?: JSONValue;
+    metadata?: object;
+  }) {
+    super(response.message);
+    this.expoApiV2ErrorCode = response.code;
+    this.expoApiV2ErrorDetails = response.details;
+    this.expoApiV2ErrorServerStack = response.stack;
+    this.expoApiV2ErrorMetadata = response.metadata;
+  }
+}
+
+interface RequestOptions {
+  body: JSONValue;
+}
+
+class ApiV2 {
+  public async putAsync(path: string, options: RequestOptions): Promise<any> {
+    return await this.requestAsync(path, { method: 'PUT', body: JSON.stringify(options.body) });
+  }
+
+  public async postAsync(path: string, options: RequestOptions): Promise<any> {
+    return await this.requestAsync(path, { method: 'POST', body: JSON.stringify(options.body) });
+  }
+
+  public async deleteAsync(path: string): Promise<any> {
+    return await this.requestAsync(path, { method: 'DELETE' });
+  }
+
+  public async getAsync(path: string): Promise<any> {
+    return await this.requestAsync(path, { method: 'GET' });
+  }
+
+  private async requestAsync(path: string, options: RequestInit): Promise<any> {
+    try {
+      const response = await fetch(`${getExpoApiBaseUrl()}/v2/${path}`, {
+        ...options,
+        headers: {
+          'Content-Type': 'application/json',
+          ...this.getAuthHeaders(),
+        },
+      });
+      return await response.json();
+    } catch (err) {
+      await this.handleApiErrorAsync(err);
+    }
+  }
+
+  private async handleApiErrorAsync(err: any): Promise<void> {
+    if (err instanceof RequestError) {
+      let result: { [key: string]: any };
+      try {
+        result = await err.response.json();
+      } catch {
+        throw new Error(`Malformed api response: ${await err.response.text()}`);
+      }
+      if (result.errors?.length) {
+        throw new ApiV2Error(result.errors[0]);
+      }
+    } else {
+      throw err;
+    }
+  }
+
+  private getAuthHeaders(): Record<string, string> {
+    const token = getAccessToken();
+    if (token) {
+      return { authorization: `Bearer ${token}` };
+    }
+    const sessionSecret = getSessionSecret();
+    if (sessionSecret) {
+      return { 'expo-session': sessionSecret };
+    }
+    return {};
+  }
+}
+
+export const api = new ApiV2();
```

Comment:
optional: alternatively this class could be all static methods since it doesn't need state